File: app/src/main/java/com/example/tribecrypto/repository/WatchListRepository.java
package com.example.tribecrypto.repository;

import com.example.tribecrypto.data.entity.WatchListEntity;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;

public interface WatchListRepository {

    Observable<List<WatchListEntity>> getAllWatchList();
    Completable insertWatchCurrency(WatchListEntity data);
    Completable insertWatchCurrencies(List<WatchListEntity> data);
    Completable updateWatchListCurrency(WatchListEntity data);
    Completable updateWatchListCurrencies(List<WatchListEntity> data);
    Completable removeWatchlistEntity(String name);
    Completable removeAllWatchList();
}

File: app/src/main/java/com/example/tribecrypto/repository/WatchListRepository.kt
package com.example.tribecrypto.repository

import com.example.tribecrypto.data.entity.WatchListEntity
import io.reactivex.Completable
import io.reactivex.Observable

interface WatchListRepository {

    fun getAllWatchList(): Observable<List<WatchListEntity>>
    fun insertWatchCurrency(data: WatchListEntity): Completable
    fun insertWatchCurrencies(data: List<WatchListEntity>): Completable
    fun updateWatchListCurrency(data: WatchListEntity): Completable
    fun updateWatchListCurrencies(data: List<WatchListEntity>): Completable
    fun removeWatchlistEntity(name: String): Completable
    fun removeAllWatchList(): Completable
}